Sources/protoc-gen-swift/MessageFieldGenerator.swift

```diff
@@ -207,13 +207,13 @@ class MessageFieldGenerator: FieldGeneratorBase, FieldGenerator {
             assert(fieldDescriptor.file.syntax == .proto3)
             switch fieldDescriptor.type {
             case .string, .bytes:
-                conditional = ("!\(varName).isEmpty")
+                conditional = ("!\(varName).isEmpty || visitor.shouldIncludeDefault()")
             default:
-                conditional = ("\(varName) != \(swiftDefaultValue)")
+                conditional = ("\(varName) != \(swiftDefaultValue) || visitor.shouldIncludeDefault()")
             }
         }
 
-        p.print("if \(conditional) {\n")
+        p.print("if \(conditional) || visitor.shouldIncludeDefault() {\n")
```

Comment:
Didn't you put the new call in conditional already, aren't you going to call in multiple times?

Reply:
oh yeah, I should create locale boolean at the beginning of traverse function